Feature/mimxrt1050 evk qspi support

ainguraXmarquiegui opened the pull request:
Add real support for MIMXRT1050_EVK_QSPI

This merge request depends on these other two:

https://github.com/zephyrproject-rtos/hal_nxp/pull/118
https://github.com/zephyrproject-rtos/mcuboot/pull/64

ainguraXmarquiegui committed:
boards: mimxrt1050_evk_qspi: Add support for QSPI flashing

Add instructions to be able to enable QSPI flash support.

Signed-off-by: Xabier Marquiegui <xmarquiegui@ainguraiiot.com>

ainguraXmarquiegui commented:
Related issue: https://github.com/zephyrproject-rtos/zephyr/issues/38665

utzig commented on boards/arm/mimxrt1050_evk/doc/index.rst:
```diff
@@ -290,6 +290,10 @@ the `OpenSDA J-Link MIMXRT1050-EVK-Hyperflash Firmware`_. Check that jumpers
 J32 and J33 are **on** (they are on by default when boards ship from the
 factory) to ensure SWD signals are connected to the OpenSDA microcontroller.
 
+Follow the instructions in :ref:`Enable QSPI flash support in SEGGER JLink`_
```
Don't use `:ref:` here, the `_` at the end links it.

ainguraXmarquiegui replied:
Fixed. Thank you for the heads up. I'm not too familiar with .rst file formatting.

ainguraXmarquiegui commented on boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts:
```diff
@@ -19,7 +19,7 @@
 	};
 
-	chosen {
+	chosen: chosen {
```
It needs a tag so that it can be referenced by `mimxrt1050_evk_qspi.dts` in order to add:

```
&chosen {
	zephyr,flash-controller = &flexspi;
	zephyr,flash = &is25wp064;
	zephyr,code-partition = &slot0_partition;
};
```

mbolivar-nordic replied:
> It needs a tag

https://docs.zephyrproject.org/latest/guides/dts/howtos.html#use-devicetree-overlays

It doesn't need a tag. See option 2 in the above link or just keep it the way it is.

ainguraXmarquiegui replied:
You are right.

carlescufi commented:
@dleach02 can you suggest more reviewers here?

mbolivar-nordic commented on boards/arm/mimxrt1050_evk/mimxrt1050_evk_qspi.dts:
```diff
@@ -8,15 +8,14 @@
-		/delete-property/ zephyr,flash;
-		/delete-property/ zephyr,code-partition;
-	};
+&chosen {
```
There is nothing wrong with `/ { chosen { ... }; };`, is there? The path to the chosen node is well known and IMO we should just refer to it by path in overlays.

ainguraXmarquiegui replied:
You are absoulutely right @mbolivar-nordic . Thank you for your comment. It's always good to learn.

mbolivar-nordic reviewed (changes requested):
I have some style nits.

ainguraXmarquiegui commented:
Apparently nicolai.glud@prevas.dk has already merged part of this work in https://github.com/zephyrproject-rtos/zephyr/commit/28a4bb076314629561e93299c894cc4a77490b97 .

I still think it would be good for any user to have this commit (which is part of this pull request) also available on mainline: https://github.com/ainguraXmarquiegui/zephyr/commit/1bde6aa5713fd892978aee5ab763efa5b8759cc5

also, this pull request is missing: https://github.com/zephyrproject-rtos/hal_nxp/pull/118

mbolivar-nordic reviewed (approved):
DT bits look fine to me, thank you very much for the update. @dleach02 @mmahadevan108 ?

dleach02 commented:
@danieldegrasse @mmahadevan108, can you give a once over?

brycewilkins commented on boards/arm/mimxrt1050_evk/doc/index.rst:
```diff
@@ -298,6 +298,10 @@ the `OpenSDA J-Link MIMXRT1050-EVK-Hyperflash Firmware`_. Check that jumpers
 
+Follow the instructions in `Enable QSPI flash support in SEGGER JLink`_
+in order to support your EVK if you have modified it to boot from QSPI NOR
+flash as specifiee by NXP AN12108.
```
Just a small typo: `specifiee` -> `specified`

ainguraXmarquiegui replied:
You are right, thank you. Fixed on my last force push.

ainguraXmarquiegui committed:
modules: hal_nxp: Update to support boot from QSPI

Update hal_nxp to add support for mimxrt1050_evk_qspi.
This board should support both boot-from-hyperflash and
boot-from-qspi. This modification enables a modified board
to boot from the on-board qspi nor chip.

Signed-off-by: Xabier Marquiegui <xmarquiegui@ainguraiiot.com>

danieldegrasse reviewed (changes requested):
@ainguraXmarquiegui, can you update `west.yml` to reflect the dependency on the PR (zephyrproject-rtos/hal_nxp#118) in NXP's HAL? [Here's documentation](https://docs.zephyrproject.org/latest/guides/modules.html#process-for-submitting-changes-to-existing-modules) on how to add that dependency, and #41171 is an example of how to do that in practice

ainguraXmarquiegui commented:
Thank you  @danieldegrasse . I have updated the file already. I hope I did it alright.

danieldegrasse reviewed (approved):
@ainguraXmarquiegui Yeah, that's all correct-thanks for making the change. Once the PR on the NXP side is merged, you can update this PR so that the `west.yml` file refers to the new commit SHA within the NXP HAL repo you would like to use. At that point the github bot will remove the DNM label, and this PR should be ready to go.

mbolivar-nordic commented on west.yml:
```diff
@@ -98,7 +98,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: 66db7c2a57ec1adfd3574169346ed3af9bf17c0b
+      revision: pull/118/head
```
This commit is just changing west.yml, but the commit log says:

```
Adds flash partitions and chosen nodes to the mimxrt1050_evk_qspi
device tree to support mcuboot on the internal QSPI flash.

Also enables FlexSPI flash driver XIP mode support on this board to
support mcuboot.
```

This doesn't seem to be doing any of that; what am I missing here? It looks like you need to fix this commit to add those DT changes.

ainguraXmarquiegui replied:
You are right. At this point that commit message no longer makes sense. It probably is a consequence of this: https://github.com/zephyrproject-rtos/zephyr/pull/38872#issuecomment-964473636 , but I should have updated the commit message.

ainguraXmarquiegui replied:
@mbolivar-nordic , please, let me know if you find this commit message to be more suitable:

https://github.com/zephyrproject-rtos/zephyr/pull/38872/commits/873494b4bd4e6a2f9d1c7d519e35296d3591e2b6
